guava/src/com/google/common/base/Converter.java

```diff
@@ -386,7 +406,19 @@ public String toString() {
   @Deprecated
   @Override
   @CanIgnoreReturnValue
-  public final @Nullable B apply(@Nullable A a) {
+  @SuppressWarnings({
+    "override.return.invalid",
+    "override.param.invalid"
+    /*
+     * This method can return null value even if only type A is declared to be nullable and null is
```

Comment:
This seems incorrect to me.  A function that returns B should not return null unless B is a nullable type.
Furthermore, if A is a non-null type, then it should not be permitted to pass null into `apply`.

Reply:
Yes, this method can return a null value even if B is not declared to be nullable. It depends upon the type of argument that is supplied. I'm not sure how to depict this behaviour using annotations. Should this be reported as a bug?

Reply:
Where does the documentation say that the method is allowed to return a different type than B?

Reply:
By adding an overriding `@PolyNull` annotation we are describing that this method may return null even if B is declared to be `@NonNull`. It depends entirely on whether a null is passed as an argument to this method or not.  Nullness Checker will expects a nullable value whenever a nullable value is passed to this method (even if B is declared to be `@NonNull`). This is conflicting with the overridden method, which expects it to be annotated as ```public final B apply (A a)```. But if we do so then we will be missing on the case when B is declared as `@NonNull` and A as `@Nullable` and this method still might return a nullable value. Is this the way we should be handling this? Please let me know your thoughts on this